common/net/tlv.py

```diff
@@ -0,0 +1,146 @@
+from common.tools import is_256b_hex_str
+
+import common.net.bigsize as bigsize
+from common.net.utils import message_sanity_checks
+
+tlv_types = {
+    "networks": b"\x01",
+    "amt_to_forward": b"\x02",
+    "outgoing_cltv_value": b"\x04",
+    "short_channel_id": b"\x06",
+    "payment_data": b"\x08",
+}
+
+
+class TLVRecord:
+    """
+    Base class for TLV records.
+
+    Args:
+        t (:obj:`bytes`): the message type.
+        l (:obj:`bytes`): the value length.
+        v (:obj:`bytes`): the message value.
+    """
+
+    def __init__(self, t=b"", l=b"", v=b""):
+        if not isinstance(t, bytes):
+            raise TypeError("t must be bytes")
+        if not isinstance(l, bytes):
+            raise TypeError("l must be bytes")
+        if not isinstance(v, bytes):
+            raise TypeError("v must be bytes")
+
+        self.type = t
+        self.length = l
+        self.value = v
+
+    def __len__(self):
+        """Returns the length of the serialised TLV record"""
+        return len(self.serialize())
+
+    def __eq__(self, other):
+        return isinstance(other, TLVRecord) and self.value == other.value
```

Comment:
Agreed. Initially I was thinking that type and length won't matter since all messages of a given type have a given length, and messages of a given subclass are forced and checked the type and length. However this does no apply for unknown types.

``return isinstance(other, TLVRecord) and self.serialize() == other.serialize()`` seems the most complete approach.